mastodon_to_sqlite/service.py

```diff
@@ -68,6 +68,8 @@ def build_database(db: Database):
                 "account_id": int,
                 "content": str,
                 "created_at": str,
+                "bookmarked": int,
+                "favourited": int,
```

Comment:
In order to support multiple Mastodon accounts in the same database, we'll need to make some changes. Adding bool fields to track bookmarks and favourites won't work for this use case, since we need to keep track of which account these actions were performed on. To solve this, I suggest creating a new table specifically for tracking bookmarks and favourites, with columns for the account ID, status ID, and type of action.

Here's some example code for creating this table:

```python
favourites_table = get_table("favourites", db=db)
if favourites_table() is False:
    favourites_table.create(
        columns={"account_id": int, "status_id": int, "type": str},
        pk=("account_id", "status_id"),
        foreign_keys=(
            ("account_id", "accounts", "id"),
            ("status_id", "statuses", "id"),
        )
    )

    favourites_indexes = {tuple(i.columns) for i in favourites_table}
    if ("account_id",) not in favourites_indexes:
        favourites_table(["account_id"])
    if ("status_id",) not in favourites_indexes:
        favourites_table(["status_id"])
```

This will allow us to connect multiple Mastodon accounts in one SQLite table and keep track of which actions were performed by which accounts. Let me know if you have any questions or concerns.

Reply:
Sounds good wrt bookmarks and favourites though I'm not sure about reblogs. Reblogs are basically nested statuses (with the original status object stored in the `reblog` key), and the information in the root status—like `created_at`, though possibly many more depending on if/how quotes get implemented—is going to be important to capture.

Not really a problem that needs to be solved for this PR though.